salt/modules/github.py

```diff
@@ -969,6 +1046,17 @@ def edit_repo(name,
 
     .. versionadded:: Carbon
     '''
+
+    try:
+        allow_private_change = _get_config_value(profile, 'allow_repo_privacy_changes')
+    except CommandExecutionError:
+        allow_private_change = True
```

Comment:
Fair point, I'll change the default to `False`. I initially wasn't sure whether it was too presumptuous to turn this off by default but it's probably best to err on the side of caution